src/main.ts

```diff
@@ -130,14 +131,23 @@ async function installCargoSemverChecks(cargo: rustCore.Cargo): Promise<void> {
     }
 }
 
+function getManifestDir(): string {
+    const manifestPath = rustCore.input.getInput("manifest-path") || "./";
+    return path.extname(manifestPath) ? path.dirname(manifestPath) : manifestPath;
+}
```

Comment:
nitpick: I would inline this function into `run()` because due to a subtlety, it actually isn't safe to call at any point outside of `run()`.

`run()` is the entry point of the action. It _knows_ what path `"./"` points to. This isn't true at any other point in the program.

I would also immediately canonicalize the path here. If the `"./"` default is used, we want to turn it into an absolute path ASAP or else we have the same problem as above.

Reply:
Good idea, thank you! 

When I inline this function into `run()`, does it still make sense to canonicalize the path here? It's being resolved in `RustdocCache` constructor.

Reply:
Yes, I'd canonicalize everything immediately at the top of `run()`. That's the only place that has a consistent idea of where the action is running. Everything else should be path-independent since it can't guarantee what working directory it's running in.